XUtils.h

```diff
@@ -117,6 +117,15 @@ int compareRealNumbers(double a, double b);
    nonnegative. */
 double sumPositiveValues(const double* array, size_t count);
 
+/* Returns the number of trailing zero bits */
+#if defined(HAVE_BUILTIN_CTZ)
+static inline unsigned int countTrailingZeros(unsigned int x) {
+   return !x ? 32 : __builtin_ctz(x);
```

Comment:
Just for information, C23 would support `stdc_trailing_zeros()` function with mostly the same semantics of `__builtin_ctz`.

```c
#ifdef HAVE_STDBIT_H
#include <stdbit.h>
// unsigned int stdc_trailing_zeros_ui(unsigned int x);
// unsigned long stdc_trailing_zeros_ul(unsigned long x);
// stdc_trailing_zeros(x) defined as type-generic macro
#endif

#ifdef HAVE_STDC_TRAILING_ZEROS
static inline unsigned int countTrailingZeros(unsigned int x) {
   return !x ? 32 : stdc_trailing_zeros_ui(x);
}
#endif
```